src/Page.js

```diff
@@ -628,8 +628,7 @@ class Page {
     this.includedFiles.add(headerPath);
     // Map variables
     const parentSite = urlUtils.getParentSiteAbsolutePath(this.sourcePath, this.rootPath, this.baseUrlMap);
-    const userDefinedVariables = this.userDefinedVariablesMap[parentSite];
-    return `${njUtil.renderRaw(headerContent, userDefinedVariables)}\n${pageData}`;
+    return `${this.variablePreprocessor.renderSiteVariables(parentSite, headerContent)}\n${pageData}`;
```

Comment:
boilerplate reduction for `site` variables ( formerly `userDefinedVariablesMap` ) which is completed in this PR